integration_tests/src/lib.rs

```diff
@@ -5809,4 +5809,41 @@ mod test {
             }
         ));
     }
+
+    #[tokio::test]
+    async fn test_auto_flush_deactivate() {
+        let log = csl();
+        let child = TestDownstairsSet::small(false).await.unwrap();
+        let vcr = VolumeConstructionRequest::Volume {
+            id: Uuid::new_v4(),
+            block_size: 512,
+            sub_volumes: vec![VolumeConstructionRequest::Region {
+                block_size: 512,
+                blocks_per_extent: child.blocks_per_extent(),
+                extent_count: child.extent_count(),
+                opts: child.opts(),
+                gen: 2,
+            }],
+            read_only_parent: None,
+        };
+
+        let volume = Volume::construct(vcr, None, log.clone()).await.unwrap();
+        volume.activate().await.unwrap();
+
+        // Send exactly IO_CACHED_MAX_JOBS to force a Barrier to be sent.  The
+        // barrier empties out the active list, so deactivation may proceed.
+        for _ in 0..10000 {
```

Comment:
nit: should this instead be `IO_CACHED_MAX_JOBS + N`

Reply:
`IO_CACHED_MAX_JOBS` isn't public in the `crucible` crate, because it's (mostly) an implementation detail.  I'm ambivalent about making it public versus hard-coding it here, let me know if you have strong opinions.

Reply:
...and it must be an exact multiple of `IO_CACHED_MAX_JOBS`, because otherwise there would be leftover jobs in `can_deactivate_immediately` and deactivation wouldn't skip the final flush.

Reply:
if it must be exactly that value, I'd say make it public or have a way to return it (less excited about that option)

Reply:
alternatively, mimic what omicron does and use

```
#[cfg(any(test, feature = "testing"))]
```

to guard if it's pub or not?

Reply:
Okay, done in 7dc0be16221d

There's not a particularly clean way to enable / disable the `pub` visibility qualifier, so I added a dedicated module that re-exports the constant.